command to set a variable in activation environment

q66 opened the pull request:
This is a pretty rough draft, mostly to get comments right now.

It adds a new `setenv` command to `dinitctl`. The command can take any number of variables, either as variable names alone (in which case the value is retrieved from the environment `dinitctl` is called in) or in the format `FOO=bar` (which encodes the value).

Limitations:

- length of the name + value + `=` is limited to 1024 minus protocol overhead characters (do we want to bother with a chunked protocol that would let one pass longer vars? i'm leaning towards not, as it seems to be of limited use and likely a sign of doing something terribly wrong)

Fixes https://github.com/davmac314/dinit/issues/43

davmac314 commented on src/dinitctl.cc:
```diff
@@ -1287,3 +1304,47 @@ static int enable_disable_service(int socknum, cpbuffer_t &rbuffer, const char *
+        write_all_x(socknum, buf.data(), bufs + 3);
+        wait_for_reply(rbuffer, socknum);
+        if (rbuffer[0] == DINIT_RP_BADREQ) {
+            if (verbose) {
```
Don't need `verbose` check for error messages, error output should always visible

davmac314 commented:
Seems fine to me. One niggle is that it would be better to do error checking in dinitctl in case an environment variable does happen to be oversize (> 1024 - 3 bytes) ; it would be better for user experience if `dinitctl` itself gave a specific error message in that case.

Support for chunking larger environment variables would be nice but as you say it's unlikely to be needed in practice.

q66 commented:
alright, i will add the documentation+tests and fix up the code then; i already deployed this in my distribution (to handle session bus address automatically with pipewire user services) and it seems to work quite well

q66 commented:
added integration test, updated manpage, added early length check; if there's anything else let me know, otherwise i think it should be baseline ready

davmac314 commented:
All looks fine.
Only one thing - do you want to add your name to the CONTRIBUTORS file?

q66 commented:
done
